pkg/reconciler/ingresssplitter/ingress.go

```diff
@@ -275,3 +276,11 @@ func createOwnedBySelector(clusterName, name, namespace string) (labels.Selector
 
 	return ownedBySelector, nil
 }
+
+func LabelEscapeClusterName(s string) string {
```

Comment:
I have a follow-up in mind (which should be easy with all the cleanup) to get rid of the colon.

The plan:
- use `/` instead of `:` both in clusterName and in `/clusters/org/workspace`
- use colon or whatever in etcd paths (Clayton mentioned problems with slash)
- keep `root` and `system:*` as the only single-word clusterNames.